src/test/regress/sql/remove_temp_schemas.sql

```diff
@@ -0,0 +1,22 @@
+DO $$
+DECLARE
+	nsp TEXT;
+BEGIN
+	FOR nsp IN
+		SELECT distinct nspname
+		FROM (
+			SELECT nspname
+			FROM gp_dist_random('pg_namespace')
+			WHERE nspname ~ '^pg(_toast)?_temp_[0-9]+'
+			UNION
+			SELECT nspname
+			FROM pg_namespace
+			WHERE nspname ~ '^pg(_toast)?_temp_[0-9]+'
+		) n
+		LEFT OUTER JOIN pg_stat_activity x on
+			sess_id = regexp_replace(nspname, 'pg(_toast)?_temp_', '')::int
+		WHERE x.sess_id is null
```

Comment:
Why `UNION` and not `UNION ALL` as in the original query, even though we do `distinct` at the end? Also, in which case can sess_id be NULL?

Reply:
We do `UNION` because we want to get all relations without duplicates (`select distinct` also filters duplicates). `x.sess_id` is NULL if there is not active session at `pg_stat_activity` with `sess_id` from `regexp_replace(nspname, 'pg(_toast)?_temp_', '')::int`.
I checked, `select distinct` is extra. I removed "distinct" from queries.